python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1906 @@
+        donutStamps = []
+
+        for i in range(numStamps):
+            # Create MaskedImageF for this donut
+            stampIm = afwImage.MaskedImageF(self.cropSize, self.cropSize)
+            stampIm.image.array = croppedImage[i]  # Use the i-th donut image
+            stampIm.setXY0(0, 0)  # Set origin
+
+            # Create linear WCS for this donut stamp using TARTS center
+            # Default to exposure center
+            centroidPosition = lsst.geom.Point2D(centerX, centerY)
+
+            # Try to use TARTS internal data center if available
+            if tartsInternalData and i < len(tartsInternalData):
+                centers = tartsInternalData[i].get('centers', None)
+                if centers is not None:
+                    centersNp = self._toNumpy(centers)
+                    if len(centersNp) >= 2:
+                        centroidPosition = lsst.geom.Point2D(
+                            float(centersNp[0]), float(centersNp[1])
+                        )
+                        self.log.debug(
+                            "Using TARTS internal center [%.2f, %.2f] for donut %d",
+                            centersNp[0], centersNp[1], i+1
+                        )
+
+            # Log the final center being used
+            if centroidPosition.getX() == centerX and centroidPosition.getY() == centerY:
+                self.log.debug(
+                    "Using exposure center [%.2f, %.2f] for donut %d", centerX, centerY, i+1
+                )
+
+            wcs = exposure.wcs
+            linearTransform = wcs.linearizePixelToSky(centroidPosition, lsst.geom.degrees)
+            cdMatrix = linearTransform.getLinear().getMatrix()
+            linearWcs = lsst.afw.geom.makeSkyWcs(
+                centroidPosition,
+                wcs.pixelToSky(centroidPosition),
+                cdMatrix
+            )
+
+            # Create DonutStamp using factory
+            donutStamp = DonutStamp.factory(
+                stamp_im=stampIm,
+                metadata=metadata,
+                index=i,  # Index into the metadata lists
+                archive_element=linearWcs
+            )
+            donutStamps.append(donutStamp)
+
+        # Create DonutStamps collection
+        donutStampsObj = DonutStamps(donutStamps)
+        self.log.debug("Created DonutStamps collection with %d individual stamp(s)", len(donutStamps))
+
+        # Set scalar visit-level metadata directly (don't call
+        # _refresh_metadata as it overrides with arrays and would
+        # convert scalar values to arrays, potentially losing data
+        # when there are no stamps)
+        # Use direct assignment to ensure scalar values are stored
+        donutStampsObj.metadata["DET_NAME"] = detectorName
+        donutStampsObj.metadata["CAM_NAME"] = cameraName
+        donutStampsObj.metadata["BANDPASS"] = bandLabel
+        donutStampsObj.metadata["DFC_TYPE"] = defocalType
+        donutStampsObj.metadata["DFC_DIST"] = 1.5
```

Comment:
why is this not using the defocal distance you calculated before and you are hardcoding it here?

Reply:
Additionally, why not populate the metadata all at once below, it will help with readability of the code, i think